src/ocm/components/clusterConfiguration/manifestsConfiguration/components/CustomManifestsArray.tsx

```diff
@@ -174,47 +174,48 @@ const Manifests = <CustomManifestType,>({
   const [ExpandedManifests, setExpandedManifests] = React.useState<ExpandedManifests>(
     getExpandedManifestsDefaultValue(field.value.length),
   );
-  const [copyConfiguration] = React.useState<boolean>(false);
-  const [manifestIdxToRemove, setmanifestIdxToRemove] = React.useState<number | null>(null);
+  const [manifestIdxToRemove, setManifestIdxToRemove] = React.useState<number | null>(null);
   const { addAlert } = useAlerts();
-  if (field.value === undefined) {
-    return <LoadingState />;
-  }
-
-  const onAddManifest = () => {
-    let newManifestData: CustomManifestType;
-    if (copyConfiguration) {
-      newManifestData = cloneDeep(field.value[field.value.length - 1]);
-    } else {
-      newManifestData = cloneDeep(emptyManifestData);
-    }
+  const onAddManifest = React.useCallback(() => {
     const newExpandedManifests = getExpandedManifestsInitialValue(field.value.length + 1);
     newExpandedManifests[field.value.length] = true;
     setExpandedManifests(newExpandedManifests);
-    push(newManifestData);
-  };
+    push(getEmptyFormViewManifest());
+  }, [field.value.length, push]);
 
-  const removeManifest = async (clusterId: string) => {
-    if (manifestIdxToRemove) {
-      const manifestToRemove = field.value[manifestIdxToRemove];
-      if (manifestToRemove['folder'] !== '' && manifestToRemove['filename'] !== '') {
-        try {
-          await ClustersAPI.removeCustomManifest(
-            clusterId,
-            manifestToRemove['folder'] as string,
-            manifestToRemove['filename'] as string,
-          );
-        } catch (e) {
-          handleApiError(e, () =>
-            addAlert({
-              title: 'Manifest could not be deleted',
-              message: getApiErrorMessage(e),
-            }),
-          );
+  const removeManifest = React.useCallback(
+    async (clusterId: string) => {
+      if (manifestIdxToRemove) {
+        const manifestToRemove = field.value[manifestIdxToRemove];
+        if (manifestToRemove['folder'] !== '' && manifestToRemove['filename'] !== '') {
+          try {
+            await ClustersAPI.removeCustomManifest(
+              clusterId,
+              manifestToRemove['folder'] as string,
+              manifestToRemove['filename'] as string,
+            );
+          } catch (e) {
+            handleApiError(e, () =>
+              addAlert({
+                title: 'Manifest could not be deleted',
+                message: getApiErrorMessage(e),
+              }),
+            );
+          }
         }
       }
-    }
-  };
+    },
+    [addAlert, field.value, manifestIdxToRemove],
+  );
+  const onConfirm = React.useCallback(async () => {
```

Comment:
I think it's better to ensure that no action happens if there is no manifest to remove.
And we can simplify the code in `removeManifest` to avoid double-checking if there is or not.

```
  const onConfirm = React.useCallback(async () => {
    if (manifestIdxToRemove !== null) {
      await removeManifest(manifestIdxToRemove);
      remove(manifestIdxToRemove);
      setManifestIdxToRemove(null);
    }
  }, [removeManifest, remove, manifestIdxToRemove, setManifestIdxToRemove]);
```

and

```
  const removeManifest = React.useCallback(
    async (manifestIdx: number) => {
      const manifestToRemove = field.value[manifestIdx];      
      ...
    },
    [addAlert, clusterId, field.value],
  );
```